test/repo-url.test.js

```diff
@@ -0,0 +1,49 @@
+'use strict'
+
+const chai = require('chai')
+const repoUrl = require('../lib/repo-url')
+
+const should = chai.should()
+
+describe('repo-url', () => {
+
+  describe('getRepoUrl', () => {
+    it('return null if package is not installed', () => {
+      should.equal(repoUrl.getRepoUrl('not-installed/package'), null)
+    })
+    it('return null repository field is undefined', () => {
+      should.equal(repoUrl.getRepoUrl('package-name', {}), null)
+    })
+    it('return null repository field is unknown type', () => {
+      should.equal(repoUrl.getRepoUrl('package-name', { repository: true }), null)
+    })
+    it('return url directly from repository field if valid github url', () => {
+      repoUrl.getRepoUrl('package-name', { repository: 'https://github.com/user/repo' }).should.equal('https://github.com/user/repo')
+    })
+    it('return url directly from repository field if valid gitlab url', () => {
+      repoUrl.getRepoUrl('package-name', { repository: 'https://gitlab.com/user/repo' }).should.equal('https://gitlab.com/user/repo')
+    })
+    it('return url directly from repository field if valid bitbucket url', () => {
+      repoUrl.getRepoUrl('package-name', { repository: 'https://bitbucket.org/user/repo' }).should.equal('https://bitbucket.org/user/repo')
+    })
+    it('return url constructed from github shortcut syntax string', () => {
+      repoUrl.getRepoUrl('package-name', { repository: 'user/repo' }).should.equal('https://github.com/user/repo/tree/master')
+    })
+    it('return url constructed from repository specific shortcut syntax string', () => {
+      repoUrl.getRepoUrl('package-name', { repository: 'github:user/repo' }).should.equal('https://github.com/user/repo/tree/master')
+    })
+    it('return url constructed from git-https protocol', () => {
+      repoUrl.getRepoUrl('package-name', { repository: { url: 'git+https://github.com/user/repo.git' } }).should.equal('https://github.com/user/repo/tree/master')
+    })
+    it('return url constructed from git protocol', () => {
+      repoUrl.getRepoUrl('package-name', { repository: { url: 'git://github.com/user/repo.git' } }).should.equal('https://github.com/user/repo/tree/master')
+    })
+    it('return url constructed from http protocol', () => {
+      repoUrl.getRepoUrl('package-name', { repository: { url: 'http://github.com/user/repo.git' } }).should.equal('https://github.com/user/repo/tree/master')
+    })
+    it('return url with directory path', () => {
+      repoUrl.getRepoUrl('package-name', { repository: { url: 'http://github.com/user/repo.git', directory: 'packages/specific-package' } }).should.equal('https://github.com/user/repo/tree/master/packages/specific-package')
+    })
```

Comment:
We should omit `/tree/master` whenever possible for a cleaner output.

Reply:
I was debating this. It seems like GitHub (and many others) are planning on renaming the default branch to something other than master [see here](https://github.com/github/renaming). While this does make the URLs less clean, I figured it'd be better to always include it than try to keep up with what the standard branch name is for each repository.

I was going for the least maintenance approach, but maybe I can run a test string through the `hosted-git-info` function to see what default branch name it spits out and use that to determine what string to remove when applicable.

Something like this maybe?

```
const commonBranchPath = hostedGitInfo.fromUrl('user/repo').browse('').match(/(\/tree\/[a-z]+)/)[0]
```

Reply:
Unless I am mistaken, we can follow the same pattern as the `repository` value itself: if it specifies a branch, use it; if it does not specify a branch, don't add anything. 

e.g. the repository for `npm-check-updates` is specified as: 

```json
  "repository": {
    "type": "git",
    "url": "git+https://github.com/raineorshine/npm-check-updates.git"
  },
```

This should generate `https://github.com/raineorshine/npm-check-updates`. There's no need to compare a branch at all. `hosted-git-info` is creating it. You may need to detect this manually if they don't have a way to detect a branch-less url.

Reply:
Yeah the library will always create the longest-form URL. So I'll detect and remove it.